DigitalLearningSolutions.Data.Tests/DataServices/PasswordDataServiceTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.Helpers;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class PasswordDataServiceTests
+    {
+        private PasswordDataService passwordDataService;
+        private UserDataService userDataService;
+
+        [SetUp]
+        public void Setup()
+        {
+            var connection = ServiceTestHelper.GetDatabaseConnection();
+            passwordDataService = new PasswordDataService(connection);
+            userDataService = new UserDataService(connection);
+        }
+
+        [Test]
+        public void Set_password_by_candidate_number_should_set_password_correctly()
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                // When
+                var password = "hashedPassword";
+                var candidateNumber = "KW1";
+                passwordDataService.SetPasswordByCandidateNumber(candidateNumber, password);
+                var result = userDataService.GetDelegateUserById(1)?.Password;
```

Comment:
Feels a bit cheeky relying on another service in a unit test... Any idea how to get around that?